dockerfiles/install_rocm_tarball.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# install_rocm_tarball.sh
+#
+# Downloads and installs ROCm from a tarball.
+# Supports nightlies, prereleases, and devreleases.
+#
+# Usage:
+#   ./install_rocm_tarball.sh <VERSION> <AMDGPU_FAMILY> [RELEASE_TYPE]
+#
+# Arguments:
+#   VERSION          - Full version string (e.g., 7.11.0a20251211)
+#   AMDGPU_FAMILY    - AMD GPU family (e.g., gfx110X-all, gfx94X-dcgpu)
+#   RELEASE_TYPE     - Release type: nightlies (default), prereleases, devreleases
+#
+# Examples:
+#   ./install_rocm_tarball.sh 7.11.0a20251211 gfx110X-all
+#   ./install_rocm_tarball.sh 7.11.0a20251211 gfx94X-dcgpu nightlies
+#   ./install_rocm_tarball.sh 7.10.0rc2 gfx110X-all prereleases
+
+set -e
```

Comment:
This could set `-u` as well to treat undefined variables as errors. See https://github.com/ROCm/TheRock/blob/main/docs/development/style_guide.md#setting-bash-modes